AmplifyPlugins/API/Tests/AWSAPIPluginTests/Core/AppSyncListProviderTests.swift

```diff
@@ -105,18 +105,22 @@ class AppSyncListProviderTests: XCTestCase {
         XCTAssertEqual(associatedField, "post")
     }
 
-    func testLoadedStateSynchronousLoadSuccess() {
+    func testLoadedStateLoadSuccess() {
         let elements = [Post4(title: "title"), Post4(title: "title")]
         let listProvider = AppSyncListProvider(elements: elements)
-        let results = listProvider.load()
-        guard case .success(let posts) = results else {
-            XCTFail("Should be .success")
-            return
+        let loadCompleted = expectation(description: "Load Completed")
+        listProvider.load { result in
+            guard case .success(let posts) = result else {
+                XCTFail("Should be .success")
+                return
+            }
+            XCTAssertEqual(posts.count, 2)
+            loadCompleted.fulfill()
         }
-        XCTAssertEqual(posts.count, 2)
+        wait(for: [loadCompleted], timeout: 1)
```

Comment:
This is fine in these tests because they are synchronous, but note that in async tests, you need to use `await waitForExpectations(...)` as it has been updated to work with Swift Concurrency, whereas `wait(...)` will block the thread.